incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -373,7 +377,10 @@ func (r *HierarchyConfigReconciler) syncParent(log logr.Logger, inst *api.Hierar
 
 	// Sync this namespace with its current parent.
 	curParent := r.Forest.Get(inst.Spec.Parent)
-	if curParent != nil && !curParent.Exists() {
+	if config.ExcludedNamespaces[inst.Spec.Parent] {
+		log.Info("The parent is an excluded namespace that should not be set as a parent", "parent", inst.Spec.Parent)
```

Comment:
Nit: you repeat the word "parent" three times here. Also, the log should record what HNC is doing/observing, e.g.:

`log.Info("Excluded namespace illegally set as parent", "parent", inst.Spec.Parent)`

Even better, for this and the message on line 385:

`log.Info("Setting ConditionActivitiesHalted: excluded namespace set as parent", "parent", inst.Spec.Parent)`

Reply:
Updated to `Setting Condition...`